sdk/core/azure-mgmt-core/azure/mgmt/core/_pipeline_client.py

```diff
@@ -47,23 +42,11 @@ def __init__(self, base_url, **kwargs):
                 raise ValueError(
                     "Current implementation requires to pass 'config' if you don't pass 'policies'"
                 )
-            kwargs["policies"] = self._default_policies(**kwargs)
+            per_call_policies = kwargs.get('per_call_policies', [])
+            per_call_policies.append(ARMAutoResourceProviderRegistrationPolicy())
```

Comment:
In `azure-core`, `per_call_policies` can be a `list` or `object`:

https://github.com/Azure/azure-sdk-for-python/blob/6edb9997e087ac410a17252379f5462a15a2639f/sdk/core/azure-core/azure/core/_pipeline_client.py#L125-L129

But here it can only be a `list`. Passing `per_call_policies` as an object causes failure in the `append` method.

Reply:
Good catch.